accepted/2021/preview-features/preview-features.md

```diff
@@ -0,0 +1,418 @@
+work at the library level isn't very high, but it's all about finding the right
+set of support infrastructure so that developers are able to be productive at
+writing generic code that can rely on numerical expressions and operators. In a
+sense, the library work can't really fully start until the runtime and compiler
+support is sufficiently bootstrapped, but it's the most important part in the
+validation of the feature set. Practically, this means a good chunk of the
+feature will only be available halfway through a release cycle, which greatly
+reduces the time window for gathering feedback from other parts of the product
+as well as from customers using preview builds of the upcoming .NET version.
+
+Sadly, preview feedback isn't always sufficient to judge how well a feature
+works because
+
+1. Many fewer people use preview builds of .NET and
+2. Many people don't use .NET previews for their production projects, thus
+   greatly reducing the breadth & depth at which previews are being exercised.
+
+This document is exploring a model where GA quality builds of .NET can contain
+features that are considered still in preview.
+
+The goal is to ensure that customers can discover those features & have an easy
+way to play with them while simultaneously being fully aware that these features
+aren't supported yet and thus might change (or be removed entirely) between
+versions of .NET, just like they could in preview builds.
+
+## Scenarios and User Experience
+
+### Enabling a preview runtime feature
+
+Maddy is building a Tye application. She reads in a blog post that the .NET
+runtime team is experimenting with a new GC mode, code named FancyGC.
+
+She wants to see how this improves the performance of her backend. She's glad to
+see that the new GC mode was shipped as part of .NET 6, so she doesn't need to
+figure out how to run a new version of .NET in her CI lab. So she creates a
+feature branch and enables the new GC mode simply by changing the service's
+project file as follows:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <EnablePreviewFeatures>True</EnablePreviewFeatures>
+  </PropertyGroup>
+
+</Project>
+```
+
+She then submits a draft PR against `main` to trigger their CI/CD pipeline which
+also runs some performance tests.
+
+### Enabling a preview language feature
+
+During lunch, Adriana discusses a design problem with one of the architects who
+mentions that the upcoming static interface members in C# might be able to solve
+this issue. Adriana does a quick web search and finds a blog post from the C#
+team that showcases how this feature might look like. She wonders how she can
+play with that. Optimistic, she copy & pastes the sample code into her project
+which immediately generates a compilation error:
+
+> The feature 'static interface members' is currently in Preview and
+> "unsupported". To use Preview features, use the 'preview' language version.
```

Comment:
Is this sufficient warning that Adriana is entering an unsupported configuration? Maybe add something like "To return to a supported configuration, change back to a supported language version."

Reply:
BTW, this is mixing up preview language version and 'enable preview features'.

Reply:
Or, maybe `<EnablePreviewFeatures>` is the master switch, that includes `<LangVersion>`

Reply:
I used the message that the compiler currently emits if language syntax is guarded by the preview language version. I think it would be good if we could change the wording to make it more consistent with the overall preview mode.